packages/host/app/lib/current-run.ts

```diff
@@ -419,11 +419,18 @@ export class CurrentRun {
         },
       }) as SingleCardDocument;
       searchData = await api.searchDoc(card);
+
       if (!searchData) {
         throw new Error(
           `bug: could not derive search doc for instance ${instanceURL.href}`,
         );
       }
+
+      if (cardType.displayName === 'Card') {
```

Comment:
Card's default `displayName` is "Card" (because it inherits it from `CardDef`) which is too vague - in this case we take the class name of the card, for example "Pet". But if displayName is set on the descended card class, then we take that.

Reply:
Also, I am unsure if `cardType` is a good name for covering `cardType.name` and `cardType.displayName` under different conditions. I chose this name because this is how we call this when shown in the interface, for example in the schema editor. Let me know if you have any thoughts around this

Reply:
The class name is optional isn't it? e.g. in the case of `export default class extends Card...`

Reply:
Oh good catch. In that case the name will be "default". I'll add a check for that

Reply:
Hm now that I'm looking at things, it looks like `CardDef` without name isn't allowed, for example I get a typecheck error when trying to do that:

<img width="669" alt="image" src="https://github.com/cardstack/boxel/assets/273660/dbd35311-3a6d-46fc-8b78-1e35ab037846">

Maybe @habdelra can comment on this too? Is it possible there can be a default export for a definition that extends from `CardDef`?

Reply:
You are correct. Part of our syntax analysis requires that all card classes be named.